memory: embedding crash "reading property 'model' of null" when @xenova/transformers fetch fails (no fallthrough to ruvector)

## Environment
- `ruflo` **3.14.1** (current `latest`/`alpha` dist-tag), `@claude-flow/cli` 3.14.1
- Node.js **24.18.0**, npm 11.16.0
- Windows 11 (26200), PowerShell
- Global install: `npm install -g ruflo@latest`, then `ruflo init` in the project

## Summary
After a clean install + `ruflo init` + `ruflo memory init`, **all semantic memory operations fail**:

```
$ ruflo memory store -k greeting -n default --value "The quick brown fox jumps over the lazy dog"
  ONNX embedder ready: 384d, SIMD: true
[INFO] Storing in default/greeting...
[ERROR] Cannot read properties of null (reading 'model')

$ ruflo memory search -q fox -n default
[ERROR] Cannot read properties of null (reading 'model')
```

`ruflo memory init`'s own self-test also reports it: `✗ Generate embedding — Cannot read properties of null` (and `Controller Registry: Activated 4, Failed 2`). The MCP `memory_store` / `memory_search` tools fail the same way. Note the embedder *does* report ready ("ONNX embedder ready: 384d, SIMD: true") moments before the crash.

## Root cause
In `@claude-flow/cli/dist/src/memory/memory-initializer.js`:

- `loadEmbeddingModel()` tries `@huggingface/transformers` then `@xenova/transformers` first. On this machine `@xenova/transformers` is selected and its `pipeline('feature-extraction', 'Xenova/all-MiniLM-L6-v2')` throws **`fetch failed`** (model download fails — even though ruvector's own model fetch/caching succeeds for the same model).
- That throw is caught by the function-level `catch`, which returns `{ success: false, error: 'fetch failed' }` **without setting the module-level `embeddingModelState`** and **without reaching the working ruvector ONNX branch later in the same `try` block**.
- `generateLocalEmbedding()` then runs:
  ```js
  if (!embeddingModelState?.loaded) { await loadEmbeddingModel(); } // returns failure, state stays null
  const state = embeddingModelState;          // null
  if (state.model && typeof state.model === 'function') { ... }   // throws: reading 'model' of null
  ```

Confirmed by calling the functions directly:
```
loadEmbeddingModel RESULT: {"success":false,"dimensions":0,"modelName":"none","error":"fetch failed"}
```
…while the bundled ruvector embedder works standalone and returns a real 384-dim non-zero vector:
```
ruvector embed OK len=384 nonzero=true   (getOptimizedOnnxEmbedder().embed('hello world'))
```

So the working fallback exists and is reachable — it's just skipped because a transformers failure aborts the whole function.

## Expected
A failure in the transformers loader should **fall through** to the next loader (ruvector ONNX), not abort `loadEmbeddingModel()`. And `generateLocalEmbedding()` should not dereference a null `embeddingModelState`.

## Proposed fix (two small edits, both verified)
1. Wrap the transformers loader block in `loadEmbeddingModel()` in `try/catch` so a loader failure continues to the ruvector / hash fallbacks instead of throwing out of the function:
   ```js
   if (pipelineFn && transformersSource) {
     try {
       const embedder = await pipelineFn('feature-extraction', 'Xenova/all-MiniLM-L6-v2');
       embeddingModelState = { loaded: true, model: embedder, tokenizer: null, dimensions: 384 };
       return { success: true, dimensions: 384, modelName: 'Xenova/all-MiniLM-L6-v2', loadTime: Date.now() - startTime };
     } catch (e) {
       // transformers fetch can fail offline/behind a proxy; fall through to ruvector ONNX
     }
   }
   ```
2. Null-guard `generateLocalEmbedding()`:
   ```js
   const state = embeddingModelState ?? { model: null, dimensions: 128, loaded: true };
   ```

With both applied, semantic search returns correct matches (e.g. `"animal jumping"` → a "fox jumps…" entry at 0.78; `"feline resting"` → a "cat naps…" entry ranked above the fox entry).

## Workaround for other users hitting this
Patch the two lines above in the installed copy (it's `node_modules/ruflo/node_modules/@claude-flow/cli/dist/src/memory/memory-initializer.js` for a global install), or ensure the `@xenova/transformers` model is available so its `pipeline()` doesn't need a network fetch.

## Secondary (minor)
`commands/memory.js` `store` reads `const namespace = ctx.flags.namespace;` with **no default**, so `memory store` without `-n` stores under a namespace literally named `undefined`.

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "claude-flow",
-  "version": "3.14.1",
+  "version": "3.14.2",
   "description": "Ruflo - Enterprise AI agent orchestration for Claude Code. Deploy 60+ specialized agents in coordinated swarms with self-learning, fault-tolerant consensus, vector memory, and MCP integration",
   "main": "dist/index.js",
   "type": "module",

--- a/ruflo/package.json
+++ b/ruflo/package.json
@@ -1,6 +1,6 @@
 {
   "name": "ruflo",
-  "version": "3.14.1",
+  "version": "3.14.2",
   "description": "Ruflo - Enterprise AI agent orchestration platform. Deploy 60+ specialized agents in coordinated swarms with self-learning, fault-tolerant consensus, vector memory, and MCP integration",
   "main": "bin/ruflo.js",
   "type": "module",

--- a/v3/@claude-flow/cli/package.json
+++ b/v3/@claude-flow/cli/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@claude-flow/cli",
-  "version": "3.14.1",
+  "version": "3.14.2",
   "type": "module",
   "description": "Ruflo CLI - Enterprise AI agent orchestration with 60+ specialized agents, swarm coordination, MCP server, self-learning hooks, and vector memory for Claude Code",
   "main": "dist/src/index.js",

--- a/v3/@claude-flow/cli/src/commands/memory.ts
+++ b/v3/@claude-flow/cli/src/commands/memory.ts
@@ -83,7 +83,9 @@ const storeCommand: Command = {
   action: async (ctx: CommandContext): Promise<CommandResult> => {
     const key = ctx.flags.key as string;
     let value = ctx.flags.value as string || ctx.args[0];
-    const namespace = ctx.flags.namespace as string;
+    // #2461: without `|| 'default'`, omitting -n stores under the literal
+    // string namespace "undefined" — silent data loss for first-time users.
+    const namespace = (ctx.flags.namespace as string) || 'default';
     const ttl = ctx.flags.ttl as number;
     const tags = ctx.flags.tags ? (ctx.flags.tags as string).split(',') : [];
     const asVector = ctx.flags.vector as boolean;
@@ -208,7 +210,9 @@ const retrieveCommand: Command = {
   ],
   action: async (ctx: CommandContext): Promise<CommandResult> => {
     const key = ctx.flags.key as string || ctx.args[0];
-    const namespace = ctx.flags.namespace as string;
+    // #2461: mirror the store-side default so `memory retrieve -k k` (no -n)
+    // looks under "default" rather than the literal namespace "undefined".
+    const namespace = (ctx.flags.namespace as string) || 'default';
 
     if (!key) {
       output.printError('Key is required');

--- a/v3/@claude-flow/cli/src/memory/memory-initializer.ts
+++ b/v3/@claude-flow/cli/src/memory/memory-initializer.ts
@@ -1733,24 +1733,42 @@ export async function loadEmbeddingModel(options?: {
     }
 
     if (pipelineFn && transformersSource) {
-      if (verbose) {
-        console.log(`Loading ONNX embedding model via ${transformersSource} (all-MiniLM-L6-v2)...`);
+      // #2461: pipelineFn() can throw with `fetch failed` on Windows behind
+      // a corporate proxy / strict firewall when transformers tries to pull
+      // the model files from the HuggingFace CDN. Without the catch, that
+      // throw escapes the outer try and aborts loadEmbeddingModel() with
+      // success=false BEFORE we reach the (working) ruvector ONNX fallback
+      // below — leaving embeddingModelState=null, which then crashes
+      // generateLocalEmbedding() with "Cannot read properties of null
+      // (reading 'model')" on every memory store / search call.
+      try {
+        if (verbose) {
+          console.log(`Loading ONNX embedding model via ${transformersSource} (all-MiniLM-L6-v2)...`);
+        }
+        const embedder = await pipelineFn('feature-extraction', 'Xenova/all-MiniLM-L6-v2');
+
+        embeddingModelState = {
+          loaded: true,
+          model: embedder,
+          tokenizer: null,
+          dimensions: 384 // MiniLM-L6 produces 384-dim vectors
+        };
+
+        return {
+          success: true,
+          dimensions: 384,
+          modelName: 'Xenova/all-MiniLM-L6-v2',
+          loadTime: Date.now() - startTime
+        };
+      } catch (err) {
+        if (verbose) {
+          console.warn(
+            `${transformersSource} pipeline init failed (${err instanceof Error ? err.message : String(err)}); ` +
+            'falling through to ruvector ONNX / agentic-flow / hash fallback.'
+          );
+        }
+        // Intentional fall-through to the next embedder branch.
       }
-      const embedder = await pipelineFn('feature-extraction', 'Xenova/all-MiniLM-L6-v2');
-
-      embeddingModelState = {
-        loaded: true,
-        model: embedder,
-        tokenizer: null,
-        dimensions: 384 // MiniLM-L6 produces 384-dim vectors
-      };
-
-      return {
-        success: true,
-        dimensions: 384,
-        modelName: 'Xenova/all-MiniLM-L6-v2',
-        loadTime: Date.now() - startTime
-      };
     }
 
     // Fallback: Check for agentic-flow ReasoningBank embeddings (v3)
@@ -1923,7 +1941,18 @@ export async function generateLocalEmbedding(text: string): Promise<{
     await loadEmbeddingModel();
   }
 
-  const state = embeddingModelState!;
+  // #2461: loadEmbeddingModel() can leave embeddingModelState null when an
+  // earlier loader (transformers fetch, ruvector init) throws and we never
+  // reach the hash-fallback assignment. Don't lie with a `!` non-null
+  // assertion — fall back to a synthetic hash-fallback state so we degrade
+  // to the deterministic 128-dim hash embedding instead of crashing the
+  // entire memory store/search path with "reading property 'model' of null".
+  const state = embeddingModelState ?? {
+    loaded: true,
+    model: null,
+    tokenizer: null,
+    dimensions: 128,
+  };
 
   // Use ONNX model if available
   if (state.model && typeof (state.model as any) === 'function') {

